src/Core/Echo.Concrete/Values/ValueType/MemoryBlockValue.cs

```diff
@@ -1,55 +1,61 @@
 using System;
 using System.Buffers.Binary;
 using System.Runtime.CompilerServices;
-using Echo.Concrete.Values.ValueType;
 using Echo.Core;
 using Echo.Core.Values;
 
-namespace Echo.Concrete.Values.ReferenceType
+namespace Echo.Concrete.Values.ValueType
 {
     /// <summary>
-    /// Represents a pointer to the beginning of a chunk of memory.
+    /// Represents a chunk of memory.
     /// </summary>
-    public class MemoryPointerValue : IPointerValue
+    public class MemoryBlockValue : IValueTypeValue, IMemoryAccessValue
     {
         private readonly Memory<byte> _memory;
         private readonly Memory<byte> _knownBitMask;
 
+        /// <summary>
+        /// Creates a new fully known memory pointer value.
+        /// </summary>
+        /// <param name="memory">The referenced memory.</param>
+        public MemoryBlockValue(Memory<byte> memory)
+        {
+            _memory = memory;
+            _knownBitMask = new Memory<byte>(new byte[memory.Length]);
```

Comment:
Might want to consider doing some array pooling here, like in IntegerNValue, but not a major concern.